### ISSUE TITLE
Supports authentication with GitHub

### ISSUE THREAD

**Issue by bmann** [bmann/microglue#2]
The server needs to be supplied with GitHub authentication credentials so it can push to public or private repos on GitHub. 

There are a couple of different ways this could work. 

Initially this only needs to support one user who runs their own copy of the server. However, even this one user may want to run multiple blogs in multiple repos. 

## Config files

As part of setup, edit config files or environment variables to include:

* GitHub username
* repo name / path — eg user/repo or organization/repo
* personal access token

## Web setup

Using a web interface provided by the server, perform OAuth authorization of the server, including selecting one or more repos. 

See https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#non-web-application-flow

**Comment by patdryburgh** [comment]
@bmann It’s important that we also support 2FA, which according to the document you linked to may require additional work outlined here: https://developer.github.com/v3/auth/#working-with-two-factor-authentication

**Comment by bmann** [comment]
Yes. I think that means you have to have a personal access token created just for this server.
